src/query/storages/common/cache/src/lib.rs

```diff
@@ -13,16 +13,99 @@
 // limitations under the License.
 
 mod cache;
-mod object;
+mod metrics;
 mod providers;
-mod settings;
-
-pub use cache::ObjectCacheProvider;
-pub use object::CachedObject;
-pub use object::CachedObjectAccessor;
-pub use providers::metrics::metrics_reset;
-pub use providers::ByPassCache;
-pub use providers::FileCache;
-pub use providers::MemoryBytesCache;
-pub use providers::MemoryItemsCache;
-pub use settings::CacheSettings;
+mod read;
+
+use std::borrow::Borrow;
+use std::hash::Hash;
+use std::sync::Arc;
+
+pub use providers::DiskCache;
+pub use providers::InMemoryBytesCacheHolder;
+pub use providers::InMemoryCacheBuilder;
+pub use providers::InMemoryItemCacheHolder;
+pub use read::CacheKey;
+pub use read::DiskCacheReader;
+pub use read::InMemoryBytesCacheReader;
+pub use read::InMemoryItemCacheReader;
+pub use read::LoadParams;
+pub use read::Loader;
+pub use read::LoaderWithCacheKey;
+
+pub trait CacheAccessor<K, V> {
```

Comment:
Prefer to move `CacheAccessor` from `lib.rs` to `cache.rs`, keep the lib.rs clean.

Reply:
:ok_hand: